java/ql/src/experimental/Security/CWE/CWE-502/RmiUnsafeDeserialization.ql

```diff
@@ -0,0 +1,65 @@
+/**
+ * @name Unsafe deserialization with RMI.
+ * @description If a registered remote object has a method that accepts a complex object,
+ *              an attacker can take advantage of the unsafe deserialization mechanism
+ *              which is used to pass parameters in RMI.
+ *              In the worst case, it results in remote code execution.
+ * @kind problem
+ * @problem.severity error
+ * @precision high
+ * @id java/unsafe-deserialization-rmi
+ * @tags security
+ *       external/cwe/cwe-502
+ */
+
+import java
+import semmle.code.java.frameworks.Rmi
+
+private class ObjectInputStream extends RefType {
```

Comment:
Could use more specific type here:
```suggestion
private class ObjectInputStream extends Class {
```

Reply:
Sure, I can use `Class` here. Do you know whether it would make the query faster or not?

Reply:
Using the most specific types is one recommendation of the [CodeQL documentation](https://codeql.github.com/docs/writing-codeql-queries/troubleshooting-query-performance/#use-specific-types).
Though I am not sure if there is actually a notable performance improvement, or how big the improvement will be, might be best to start a [discussion](https://github.com/github/codeql/discussions) if you want to get more detailed information from the maintainers.